src/execution.cc

```diff
@@ -118,12 +118,13 @@ bool TargetNamesToNodes(const State* state, const std::vector<std::string>& name
 
 }  // namespace
 
-Execution::Execution() : Execution(NULL, Options()) {}
-
+Execution::Execution() : Execution(NULL, Options(), std::make_unique<LoggerBasic>()) {}
 Execution::Execution(const char* ninja_command, Options options) :
+  Execution(ninja_command, options, std::make_unique<LoggerBasic>()) {}
```

Comment:
You're almost certainly going to get pushback from the ninja maintainers to use std::unique_ptr on anything that isn't *already* using it, because there are some unspecified systems that the Ninja maintainers want to preserve backwards compatibility for, though they won't explain which platforms those systems have, or really talk about it in a meaningful way.

The places where unique_ptr is allowed are the places where the code in question represents some kind of stand-alone tool, so requiring C++11 or newer on that new code was considered OK.